python/lsst/analysis/tools/actions/vector/vectorActions.py

```diff
@@ -101,128 +110,63 @@ def __call__(self, data: KeyedData, **kwargs) -> Vector:
+        doc="Suffix to add to fluxType to specify the uncertainty column", default="Err"
     )
-    band = Field[str](doc="Band to calculate the S/N in.", default="i")
-
-    def getInputSchema(self) -> KeyedDataSchema:
-        yield (fluxCol := self.fluxType), Vector
-        yield f"{fluxCol}{self.uncertaintySuffix}", Vector
-
-    def __call__(self, data: KeyedData, **kwargs) -> Vector:
-        """Computes S/N in self.fluxType
-
-        Parameters
-        ----------
-        df : `Tabular`
-
-        Returns
-        -------
-        result : `Vector`
-            Computed signal-to-noise ratio.
-        """
-        fluxCol = self.fluxType.format(**(kwargs | dict(band=self.band)))
-        errCol = f"{fluxCol}{self.uncertaintySuffix.format(**kwargs)}"
-        result = cast(Vector, data[fluxCol]) / data[errCol]  # type: ignore
-
-        return np.array(cast(Vector, result))
-
-
-class ConstantValue(VectorAction):
-    """Return a constant scalar value."""
-
-    value = Field[float](doc="A single constant value", optional=False)
-
-    def getInputSchema(self) -> KeyedDataSchema:
-        return ()
-
-    def __call__(self, data: KeyedData, **kwargs) -> Vector:
-        return np.array([self.value])
-
-
-class SubtractVector(VectorAction):
-    """Calculate (A-B)."""
-
-    actionA = ConfigurableActionField[VectorAction](doc="Action which supplies vector A")
-    actionB = ConfigurableActionField[VectorAction](doc="Action which supplies vector B")
 
     def getInputSchema(self) -> KeyedDataSchema:
-        yield from self.actionA.getInputSchema()  # type: ignore
-        yield from self.actionB.getInputSchema()  # type: ignore
+        yield self.fluxType, Vector
+        yield f"{self.fluxType}{self.uncertaintySuffix}", Vector
 
     def __call__(self, data: KeyedData, **kwargs) -> Vector:
-        vecA = self.actionA(data, **kwargs)  # type: ignore
-        vecB = self.actionB(data, **kwargs)  # type: ignore
-        return vecA - vecB
-
-
-class DivideVector(VectorAction):
-    """Calculate (A/B)"""
-
-    actionA = ConfigurableActionField[VectorAction](doc="Action which supplies vector A")
-    actionB = ConfigurableActionField[VectorAction](doc="Action which supplies vector B")
-
-    def getInputSchema(self) -> KeyedDataSchema:
-        yield from self.actionA.getInputSchema()  # type: ignore
-        yield from self.actionB.getInputSchema()  # type: ignore
+        signal = np.array(data[self.fluxType.format(**kwargs)])
+        noise = np.array(data[f"{self.fluxType}{self.uncertaintySuffix}".format(**kwargs)])
+        sn = signal / noise
 
-    def __call__(self, data: KeyedData, **kwargs) -> Vector:
-        vecA = self.actionA(data, **kwargs)  # type: ignore
-        vecB = self.actionB(data, **kwargs)  # type: ignore
-        return vecA / vecB
+        return np.array(sn)
 
 
-class LoadVector(VectorAction):
-    """Load and return a Vector from KeyedData."""
+class ConvertFluxToMags(VectorAction):
```

Comment:
Either this should be named `ConvertFluxToMag` or `ConvertFluxesToMags`. I prefer the singular myself.